src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -7,8 +7,120 @@ and parameters as expect.
 import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../../validation_test.js';
 
+// Class that indicate how to call a draw function
+class DrawCall {
```

Comment:
This class is pretty big for pretty simple behavior, and it isn't well-encapsulated so it's not very useful to write as a class (as seen by the fact that the fields aren't even initialized from the constructor, this class is pretty much being used as an `interface` or simple `{ key: value }` type).

Please consider either:
- Moving the other definition of `DrawCall` (from `robust_access_vertex.spec.ts`) into a common file, if that would work well, or
- (probably better) just have a standalone helper that takes only the necessary arguments, something like:

```ts
function callDraw(
  t: GPUTest,
  encoder: GPURenderEncoderBase,
  drawType: 'drawIndexed' | 'drawIndexedIndirect',
  { indexCount }: { indexCount: number }
) {
```

The `generateIndexedIndirectBuffer` helper is sufficiently simple that I think it could be inline in this helper, but you could also consider deduplicating it with the one in `robust_access_vertex.spec.ts`, and moving both of those helpers to `GPUTest`.

Reply:
Currently I have a callDrawIndexed to deal with drawIndexed and drawIndexedIndirect, and in the next step I will write something like callDraw to deal with draw and drawIndirect.  Test for vertex buffer OOB may need to test all these four draw calls.